docker/Dockerfile.cpu

```diff
@@ -1,22 +1,22 @@
-FROM ubuntu:16.04
+FROM ubuntu:18.04
 
 # Python3 and native deps for bindings (pyosmium, rtree).
 #
 # libsm6 native dep. is required for opencv-contrib-python.
 # See: https://github.com/skvark/opencv-python/issues/90
 RUN apt-get update -qq && \
-    apt-get install -qq -y -o quiet=1 \
-    python3 python3-dev python3-tk python3-pip build-essential libboost-python-dev libexpat1-dev zlib1g-dev libbz2-dev libspatialindex-dev libsm6
+    DEBIAN_FRONTEND=noninteractive apt-get install -qq -y -o quiet=1 \
+    python3 python3-dev python3-tk python3-pip build-essential cmake libboost-python-dev libexpat1-dev zlib1g-dev libbz2-dev libspatialindex-dev libsm6
 
 WORKDIR /app
 ADD . /app
 
-ARG TCH=0.4.0
-ARG CU=91
-ARG PY=35
+ARG TCH=1.0.1.post2
+ARG PY=36
 
+# The binary wheels for osmium lead to segfaults; compile from source
 RUN python3 -m pip install http://download.pytorch.org/whl/cpu/torch-${TCH}-cp${PY}-cp${PY}m-linux_x86_64.whl     && \
-    python3 -m pip install -r deps/requirements-lock.txt
+    python3 -m pip install -r deps/requirements-lock.txt --no-binary osmium
```

Comment:
@lonvia I'm encountering segfaults when using the osmium wheels; building from source works. I think it may be because pyosmium bundles a few libraries (I've seen a custom libz at least), which may then clash.

I'm not entirely sure who's to blame here since we're using quite a few wheels (numpy, rasterio, pytorch, etc.), what the exact cause is, and how to debug it further. Just wanted to flag it for you. Any ideas?

If you build the cpu dockerfile and remove this flag the segfault is reproducible running `./rs` in the container.

Reply:
Please look at the core dump to find out what segfaults. [This comment](https://github.com/osmcode/pyosmium/issues/87#issuecomment-449502725) has some hints on how to do that (enable core dumping with `ulimit -c unlimited`, if it doesn't dump). Also make sure you are not running into the same problem as [described in the issue](https://github.com/osmcode/pyosmium/issues/87#issuecomment-466769969).

Reply:
@daniel-j-h 

If it could be of any help,  following `rsp extract` works with RoboSat.pink and PyTorch 1.0.1 with Cuda 10.

Install from scratch on Ubuntu 18.04:
```
sudo sh -c "apt update && apt install -y build-essential python3-pip"
pip3 install git+https://github.com/datapink/robosat.pink && export PATH=$PATH:~/.local/bin
pip3 install https://download.pytorch.org/whl/cu100/torch-1.0.1.post2-cp36-cp36m-linux_x86_64.whl
wget http://us.download.nvidia.com/XFree86/Linux-x86_64/418.43/NVIDIA-Linux-x86_64-418.43.run 
sudo sh NVIDIA-Linux-x86_64-418.43.run -a -q --ui=none

wget --show-progress -q -O lyon.pbf http://datapink.tools/rsp/it/lyon.pbf
rsp extract --type Building lyon.pbf osm_lyon_footprint.json
```


`pip 3 list` post install:
```
affine (2.2.2)
albumentations (0.2.2)
asn1crypto (0.24.0)
attrs (19.1.0)
Automat (0.6.0)
blinker (1.4)
certifi (2019.3.9)
chardet (3.0.4)
Click (7.0)
click-plugins (1.0.4)
cligj (0.5.0)
cloud-init (18.4)
cloudpickle (0.8.1)
colorama (0.3.7)
command-not-found (0.3)
configobj (5.0.6)
constantly (15.1.0)
cryptography (2.1.4)
cycler (0.10.0)
dask (1.1.5)
decorator (4.4.0)
distro-info (0.18)
geojson (2.4.1)
httplib2 (0.9.2)
hyperlink (17.3.1)
idna (2.8)
imgaug (0.2.6)
incremental (16.10.1)
Jinja2 (2.10)
jsonpatch (1.16)
jsonpointer (1.10)
jsonschema (2.6.0)
keyring (10.6.0)
keyrings.alt (3.0)
kiwisolver (1.0.1)
language-selector (0.1)
MarkupSafe (1.0)
matplotlib (3.0.3)
mercantile (1.0.4)
netifaces (0.10.4)
networkx (2.2)
numpy (1.16.2)
oauthlib (2.0.6)
opencv-python-headless (4.0.0.21)
osmium (2.15.2)
PAM (0.4.2)
Pillow (5.4.1)
pip (9.0.1)
psycopg2-binary (2.7.7)
pyasn1 (0.4.2)
pyasn1-modules (0.2.1)
pycrypto (2.6.1)
pygobject (3.26.1)
PyJWT (1.5.3)
pyOpenSSL (17.5.0)
pyparsing (2.3.1)
pyproj (2.1.2)
pyserial (3.4)
python-apt (1.6.3+ubuntu1)
python-dateutil (2.8.0)
python-debian (0.1.32)
PyWavelets (1.0.2)
pyxdg (0.25)
PyYAML (3.12)
rasterio (1.0.22)
requests (2.21.0)
requests-unixsocket (0.1.5)
RoboSat.pink (0.3.5)
scikit-image (0.14.2)
scipy (1.2.1)
SecretStorage (2.3.1)
service-identity (16.0.0)
setuptools (40.8.0)
Shapely (1.6.4.post2)
six (1.12.0)
snuggs (1.4.3)
ssh-import-id (5.7)
supermercado (0.0.5)
systemd-python (234)
toml (0.10.0)
toolz (0.9.0)
torch (1.0.1.post2)
torchvision (0.2.2.post3)
tqdm (4.31.1)
Twisted (17.9.0)
ufw (0.35)
unattended-upgrades (0.1)
urllib3 (1.24.1)
webcolors (1.8.1)
wheel (0.30.0)
zope.interface (4.3.2)
```